librocksdb_sys/crocksdb/c.cc

```diff
@@ -2029,10 +2044,11 @@ void crocksdb_options_set_wal_bytes_per_sync(crocksdb_options_t* opt,
 static BlockBasedTableOptions* get_block_based_table_options(
     crocksdb_options_t* opt) {
   if (opt && opt->rep.table_factory != nullptr) {
-    void* table_opt = opt->rep.table_factory->GetOptions();
+    BlockBasedTableOptions* table_opt =
```

Comment:
Is it safe to assume it is `BlockBasedTableOptions`?

Reply:
`GetOptions` will handle the dynamic typing.

Reply:
How? As far as I know, C++ template works at compile time.

Reply:
Each type has a unique string id, each string id globally maps to an instance that can be hot loaded.